### ISSUE TITLE
Doesn't Support SAS Authentication

### ISSUE THREAD

**Issue by agilbert201** [Azure/azure-sdk-for-php#733]
This makes this library nearly unusable from a perspective of integration scenarios where you are trying to work across vendors or companies. Given that this is a potentially major use case for queues, this is glaring. We got burned, had to redo our work and went to just plain curl / php guzzle. Would have been good to at least now this up front.

**Comment by boskee** [comment]
You can use Zend to do that: http://framework.zend.com/manual/1.12/en/zend.service.windowsazure.storage.blob.html

**Comment by agilbert201** [comment]
Indeed, if you use a Shared Access Signature you can pretty much use any curl like interface, as you don't then have to sign requests. So we ended up using guzzle. But it was rather frustrating to even bother using this lib only to discover it didn't support SAS.

**Comment by dkarlovi** [comment]
This is a problem as SAS is a way for you to enable remote (Javascript) clients to write directly to the storage.

**Comment by yaqiyang** [comment]
This could be fixed after we replace PEAR packages.

**Comment by SteveAATW** [comment]
I've hit the same issue trying to use service bus queues via the PHP SDK. ACS strings are no longer generated in the Azure portal and although I've found a few referenced articles mentioning that new SB queues can be created with ACS connection strings via Azure Powershell I hit a dead end trying to accomplish this.

**Comment by yaqiyang** [comment]
Yes, you need to do this in PowerShell for the present time. 
Make sure you have Azure PowerShell installed, then in a PowerShell command prompt, run 
Add-AzureAccount # this will sign you in
New-AzureSBNamespace -CreateACSNamespace $true -Name ‘mytestbusname' -Location ‘West US’ -NamespaceType 'Messaging'

**Comment by synesthesia** [comment]
Major issue for me - I had started creating an interface to talk to an existing service bus namespace that is of the "modern" type, i.e. no ACS.  Going to end up with one namespace for most functions and a separate one just for a PHP source...

**Comment by alcohol** [comment]
Can this get some traction?

**Comment by sergey-shandar** [comment]
@alcohol We are working on a major upgrade of Azure SDK for PHP which will use [AutoRest](https://github.com/azure/autorest) to generate PHP code from Swagger.

**Comment by dkarlovi** [comment]
@sergey-shandar does this mean AutoRest will get full PHP support?

**Comment by smpilo** [comment]
This has become a critical issue with some of our partners who are using this SDK to access service bus, since Microsoft no longer allows creation of ACS namespaces. Any progress here?

**Comment by sergey-shandar** [comment]
@salameer

**Comment by smpilo** [comment]
Hi, update please?

**Comment by smpilo** [comment]
With ACS being removed from support soon, we will be unable to use this SDK soon. Is there an alternative available or coming?

**Comment by smpilo** [comment]
@salameer , @sergey-shandar 

I just re-tested with the latest sdk and it works with shared access keys, so we are OK. You can probably close this issue.
